packages/jellyfish-api-core/__tests__/category/loan/depositToVault.test.ts

```diff
@@ -0,0 +1,191 @@
+    const priceFeeds = [
+      { token: 'DFI', currency: 'USD' },
+      { token: 'BTC', currency: 'USD' },
+      { token: 'TSLA', currency: 'USD' }
+    ]
+    const oracleId = await tGroup.get(0).rpc.oracle.appointOracle(addr, priceFeeds, { weightage: 1 })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+    const timestamp = Math.floor(new Date().getTime() / 1000)
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '1@DFI', currency: 'USD' }] })
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '10000@BTC', currency: 'USD' }] })
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '2@TSLA', currency: 'USD' }] })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    // collateral token
+    await tGroup.get(0).rpc.loan.setCollateralToken({
+      token: 'DFI',
+      factor: new BigNumber(1),
+      priceFeedId: oracleId
+      // activateAfterBlock: 130  // <- hit socket hang up
+    })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    await tGroup.get(0).rpc.loan.setCollateralToken({
+      token: 'BTC',
+      factor: new BigNumber(1),
+      priceFeedId: oracleId
+    })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    // loan token
+    await tGroup.get(0).rpc.loan.setLoanToken({
+      symbol: 'TSLA',
+      priceFeedId: oracleId
+    })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    // loan scheme set up
+    await tGroup.get(0).rpc.loan.createLoanScheme({
+      minColRatio: 150,
+      interestRate: new BigNumber(3),
+      id: 'scheme'
+    })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    const vaultId = await tGroup.get(0).rpc.loan.createVault({
+      ownerAddress: await tGroup.get(0).generateAddress(),
+      loanSchemeId: 'scheme'
+    })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    return { vaultId, collateralAddress }
+  }
+
+  it('should be failed as first deposit must be DFI', async () => {
+    const promise = tGroup.get(0).rpc.loan.depositToVault({
+      id: vId, from: collateralAddr, amount: '1@BTC'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('First deposit must be in DFI')
+  })
+
+  it('should be failed as insufficient fund', async () => {
+    const promise = tGroup.get(0).rpc.loan.depositToVault({
+      id: vId, from: collateralAddr, amount: '99999@DFI'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow(`Insufficient funds: can't subtract balance of ${collateralAddr}: amount 20000.00000000 is less than 99999.00000000`)
+  })
+
+  it('should be failed as different auth address', async () => {
+    const addr = await tGroup.get(1).generateAddress()
```

Comment:
I guess here is the only reason why you setup multi containers. Probably you can change this line to `const addr = GenesisKeys[0].owner.address`. No need to setup multi containers

Reply:
it doesn't matter actually..

Reply:
gonna another node to trigger `incorrect auth`

Reply:
I cloned your code and reduced the code to use 1 container only. All tests passed. See https://github.com/DeFiCh/jellyfish/pull/665/files#diff-e06464e8721426786efb749e895e2ea38cce2135106867758b34d5023a7e7520R1

Reply:
i see.. because the node takes the addr as outsider...
but is that critical to change it to single container??

Reply:
I think he means for the simplicity. But you can add a complex test like 
```
node0 creates the vault.
node1 deposit to that vault using its funds.
```
to make multinode setup a required😉

Reply:
another test to trigger diff auth
eg: nodeAlice try deposit on nodeBob
```ts
it('should be failed as deposit by other node', async () => {
    const promise = tGroup.get(1).rpc.loan.depositToVault({
      id: vaultId, from: collateralAddress, amount: '300@DFI'
    })
    await expect(promise).rejects.toThrow(RpcApiError)
    await expect(promise).rejects.toThrow(`Incorrect authorization for ${collateralAddress}`)
  })
```

Reply:
Have you pushed the code?

Reply:
pushed